src/custom/utils/metadata.ts

```diff
@@ -0,0 +1,51 @@
+export enum MetadataKind {
+  REFERRAL = 'referrer',
+}
+
+interface Metadata {
+  kind: MetadataKind
+  version: string
+}
+
+export interface ReferralMetadata extends Metadata {
+  referrer: string
+}
+
+export type MetadataDoc = {
+  [MetadataKind.REFERRAL]?: ReferralMetadata
+}
+
+export type AppDataDoc = {
+  version: string
+  appCode?: string
+  metadata: MetadataDoc
+}
+
+export const DEFAULT_APP_CODE = 'CowSwap'
+
+export function generateReferralMetadataDoc(
+  referralAddress: string,
+  appDataDoc: AppDataDoc = generateAppDataDoc()
+): AppDataDoc {
+  return {
+    ...appDataDoc,
+    metadata: {
+      ...appDataDoc.metadata,
+      referrer: {
+        kind: MetadataKind.REFERRAL,
+        referrer: referralAddress,
+        version: '1.0.0',
```

Comment:
Alex added a [comment](https://docs.google.com/document/d/1PsdWn9WPFf2BNeoJNtClyra8jNvWk_bJ-g33TbpUf8s/edit#) suggesting removing the `version` in the Referral Metadata object:

> if it is just the referral field, we don't need a version. WHile i really don't think we need to update the referrer with more information in the future, if we want to do so, we can do it via the outer verision number

@alfetopito what do we do?

Reply:
I'm in favor of keeping it.
It might not make sense for the referrer doc right now, but there might be other docs where it does.
And then we'll have multiple docs with different standards.

@josojo I'm stalling until @anxolin is back and can convince you =P